docs/development.md

```diff
@@ -0,0 +1,342 @@
+   - Use `snake_case` for functions and variables
+   - Use `PascalCase` for types and traits
+   - Use `SCREAMING_SNAKE_CASE` for constants
+
+2. **Error Handling**:
+   - Use `Result<T, E>` for recoverable errors
+   - Use `panic!` only for unrecoverable errors
+   - Provide meaningful error messages
+
+3. **Documentation**:
+   - Add doc comments (`///`) for public APIs
+   - Include examples in documentation
+   - Document panics and errors
+
+### Example Code Style
+
+```rust
+/// Manages connector lifecycle in OpenCTI
+///
+/// # Examples
+///
+/// ```
+/// let manager = ConnectorManager::new(config)?;
+/// manager.start()?;
+/// ```
+pub struct ConnectorManager {
+    config: ManagerConfig,
+    orchestrator: Box<dyn Orchestrator>,
+}
+
+impl ConnectorManager {
+    /// Creates a new connector manager
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if configuration is invalid
+    pub fn new(config: ManagerConfig) -> Result<Self, ManagerError> {
+        // Implementation
+    }
+}
+```
+
+## Testing with Local Services
+
+### Running Local OpenCTI
+
+```bash
+# Clone OpenCTI
+git clone https://github.com/OpenCTI-Platform/opencti.git
+cd opencti/docker
+
+# Start OpenCTI stack
+docker-compose up -d
+
+# OpenCTI will be available at http://localhost:4000
+# Default credentials: admin@opencti.io / admin
+```
+
+### Running Local Docker Registry
+
+```bash
+# Start local registry for testing
+docker run -d -p 5000:5000 --name registry registry:2
+
+# Tag and push test images
+docker tag your-image localhost:5000/your-image
+docker push localhost:5000/your-image
+```
+
+## Debugging
+
+
+### Using IntelliJ IDEA
+
+1. **Install IntelliJ Rust Plugin**:
+   - Open IntelliJ IDEA
+   - Go to `File > Settings` (or `IntelliJ IDEA > Preferences` on macOS)
+   - Navigate to `Plugins`
+   - Search for "Rust" and install the official JetBrains Rust plugin
+   - Restart IntelliJ IDEA
+
+2. **Open the Project**:
+   - Use `File > Open` and select the `xtm-composer` directory
+   - IntelliJ will automatically detect the Cargo.toml file
+
+3. **Using .run Configuration Files**:
+   
+   The project already includes IntelliJ run configurations in the `dev/` directory:
+   - `dev/Run xtm-composer.run.xml` - For running the application
+   - `dev/Test xtm-composer.run.xml` - For running tests
+   
+   These should be automatically detected by IntelliJ.
+
+4. **Set Breakpoints and Debug**:
+   - Click in the gutter next to line numbers to set breakpoints
+   - Use `Run > Debug 'Debug XTM Composer'` or click the debug icon
+   - Use the debug panel to step through code, inspect variables, etc.
+
+5. **Additional IntelliJ Settings**:
+   - Enable format on save: `Settings > Tools > Actions on Save > Reformat code`
+   - Configure Rust fmt: `Settings > Languages & Frameworks > Rust > Rustfmt`
+   - Enable Clippy: `Settings > Languages & Frameworks > Rust > External Linters`
+
+### Common Issues
+
+#### 1. Connection Refused
```

Comment:
I don't see the hostname configuration in this doc, i think we should add it. Maybe just for kubernetes ?